src/go/k8s/controllers/redpanda/cluster_controller_test.go

```diff
@@ -89,6 +90,18 @@ var _ = Describe("RedPandaCluster controller", func() {
 					validOwner(redpandaCluster, svc.OwnerReferences)
 			}, timeout, interval).Should(BeTrue())
 
+			By("Creating NodePort Service")
+			Eventually(func() bool {
+				err := k8sClient.Get(context.Background(), types.NamespacedName{
+					Name:		key.Name + "-external",
+					Namespace:	key.Namespace,
+				}, &svc)
+				return err == nil &&
+					svc.Spec.Type == corev1.ServiceTypeNodePort &&
+					svc.Spec.Ports[0].Port == kafkaPort+1 &&
```

Comment:
we should document this, perhaps also in the CRD so it's visible to the user. I would like to have a follow-up issue to make this configurable

Reply:
Also what will happen if that node is not available? we'll be stuck in reconciliation?

Reply:
I disagree with the statement that the external port defined in pod spec + service spec should be configurable. This port should not be used inside the cluster. The port is not know to the external user. The NodePort service will try to create unique port which will not be configurable. The container runtime will map unique host port to the `kafkaPort+1`. 

I can only think that we should guard in webhook situation where admin or rpc port are one number above kafka port. I will add check in webhook.

> Also what will happen if that node is not available? we'll be stuck in reconciliation?

I don't follow you throughs. If node is not available then kubernetes scheduler will not assign POD to this node. Operator will not be stuck in reconciliation by the unavailability of a node.

Reply:
sorry, I wanted to say port, not node

Reply:
> Also what will happen if that PORT is not available? we'll be stuck in reconciliation?

The answer is the sts will not be created.